UnityProject/Assets/Scripts/Items/Tool/AdvancedLightReplacer.cs

```diff
@@ -85,17 +85,24 @@ private void AddLightToFixture(ItemSlot target, LightSource source, HandApply in
 		private void LightTunerWindowOpen(NetworkConnection target)
 		{
 			UIManager.Instance.GlobalColorPicker.CurrentColor = currentColor;
-			UIManager.Instance.GlobalColorPicker.EnablePicker(SetColorToTune);
+			UIManager.Instance.GlobalColorPicker.EnablePicker(SetColorToTuneWapper);
 		}
 
 		[Command(requiresAuthority = false)]
-		private void SetColorToTune(Color newColor)
+		private void SetColorToTune(Color newColor, NetworkConnectionToClient sender = null)
 		{
+			if (sender == null) return;
+			if (Validations.CanApply(PlayerList.Instance.Get(sender).Script, this.gameObject, NetworkSide.Server, false, ReachRange.Standard) == false) return;
 			if (gameObject.PickupableOrNull().ItemSlot == null) return;
 			if (gameObject.PickupableOrNull().ItemSlot.Player == null) return;
 			currentColor = newColor;
 		}
 
+		private void SetColorToTuneWapper(Color newColor)
```

Comment:
Did you mean Wrapper? Since it is just wrapping a single line, can't the security be added to the method itself?

Reply:
The validation is in the method itself, then what's the wrapper for?

Reply:
it's for the UIManager.Instance.GlobalColorPicker.EnablePicker(SetColorToTuneWapper); because it doesn't match the parameters  of the action